packages/camera/camera_avfoundation/ios/Classes/FLTCamMediaSettings.m

```diff
@@ -0,0 +1,33 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "FLTCamMediaSettings.h"
+
+@implementation FLTCamMediaSettings
+
+NS_INLINE void AssertPositiveNumberOrNil(id param, const char *paramName) {
+  if (param != nil) {
+    NSCAssert([param isKindOfClass:[NSNumber class]], @"%s is not a number: %@", paramName, param);
+    NSCAssert(!isnan([param doubleValue]), @"%s is NaN", paramName);
+    NSCAssert([param doubleValue] > 0, @"%s is not positive: %@", paramName, param);
+  }
+}
+
+- (instancetype)initWithFramesPerSecond:(nullable NSNumber *)framesPerSecond
+                           videoBitrate:(nullable NSNumber *)videoBitrate
+                           audioBitrate:(nullable NSNumber *)audioBitrate
+                            enableAudio:(BOOL)enableAudio {
+  AssertPositiveNumberOrNil(framesPerSecond, "framesPerSecond");
+  AssertPositiveNumberOrNil(videoBitrate, "videoBitrate");
+  AssertPositiveNumberOrNil(audioBitrate, "audioBitrate");
```

Comment:
Why does this helper take an `id` and assert `isKindOfClass` when the parameters are all `NSNumber`s? If anything here can be anything other than an `NSNumber`, then the signature of the init method is wrong.

Reply:
Initially, `AssertPositiveNumberOrNil` was transformed from `#define` and was intended for little bit more purposes.
Finally, it is defined as `NS_INLINE void AssertPositiveNumberOrNil(NSNumber * _Nullable param, const char * _Nonnull paramName)` and `isKindOfClass ` is removed.